src/config.c

```diff
@@ -803,15 +804,29 @@ void configGetCommand(client *c) {
     serverAssertWithInfo(c,o,sdsEncodedObject(o));
 
     for (standardConfig *config = configs; config->name != NULL; config++) {
-        if (stringmatch(pattern,config->name,1)) {
-            addReplyBulkCString(c,config->name);
-            addReplyBulkSds(c, config->interface.get(config->data));
-            matches++;
-        }
-        if (config->alias && stringmatch(pattern,config->alias,1)) {
-            addReplyBulkCString(c,config->alias);
-            addReplyBulkSds(c, config->interface.get(config->data));
-            matches++;
+        /* Hidden configs require an exact match (not a pattern) */
+        if (config->flags & HIDDEN_CONFIG) {
+            if (!strcasecmp(pattern, config->name)) {
+                addReplyBulkCString(c, config->name);
+                addReplyBulkSds(c, config->interface.get(config->data));
+                matches++;
+                break;
+            } else if (config->alias && strcasecmp(pattern, config->alias)) {
+                addReplyBulkCString(c, config->alias);
+                addReplyBulkSds(c, config->interface.get(config->data));
+                matches++;
+                break;
+            }
+        } else {
+            if (stringmatch(pattern,config->name,1)) {
+                addReplyBulkCString(c,config->name);
+                addReplyBulkSds(c, config->interface.get(config->data));
+                matches++;
+            } else if (config->alias && stringmatch(pattern,config->alias,1)) {
```

Comment:
This is slightly different, in that "config get *" currently shows both the normal name and the alias. It's come up in the past that we shouldn't just show the normal name.

Reply:
@madolson you're saying that we shouldn't have an `else if`, just another `if`, right?

Reply:
Ok, I thought it was a bug so I changed it to `else if`. The reasoning being that there's no way of knowing if your pattern matched two configs or just a config and its alias. This can be confusing.
Please confirm you want me to revert this change.

Reply:
i'm not aware of the previous discussion on that topic but:
1. if someone is doing `config get *` and then searches for either one of the aliases, we probably want him to succeed.
2. this is how it used to be till now, and this PR is about a different topic.

Reply:
ok, fixed.